library/proc_macro/src/bridge/server.rs

```diff
@@ -143,6 +144,33 @@ pub trait ExecutionStrategy {
     ) -> Buffer;
 }
 
+thread_local! {
+    /// While running a proc-macro with the same-thread executor, this flag will
+    /// be set, forcing nested proc-macro invocations (e.g. due to
+    /// `TokenStream::expand_expr`) to be run using a cross-thread executor.
+    ///
+    /// This is required as the thread-local state in the proc_macro client does
+    /// not handle being re-entered, and will invalidate all `Symbol`s when
+    /// entering a nested macro.
+    static RUNNING_SAME_THREAD: Cell<bool> = Cell::new(false);
```

Comment:
```suggestion
    static ALREADY_RUNNING_SAME_THREAD: Cell<bool> = Cell::new(false);
```
(not sure on the name but trying to have it make sense even without context)